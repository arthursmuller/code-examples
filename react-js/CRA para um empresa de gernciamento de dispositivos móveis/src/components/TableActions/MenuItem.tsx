import { MenuItem as MenuItemChakra, MenuItemProps as MenuItemPropsChakra } from '@chakra-ui/react';
import React from 'react';

interface MenuItemProps2 extends MenuItemPropsChakra {
  text: React.ReactNode;
  onClick: () => void;
}

const MenuItem = ({text, onClick, ...rest}: MenuItemProps2) => {
  return (
    <MenuItemChakra
      color="black.500"
      fontSize="sm"
      {...rest}
      onClick={onClick}
    >
      {text}
    </MenuItemChakra>
  );
}

export default MenuItem;