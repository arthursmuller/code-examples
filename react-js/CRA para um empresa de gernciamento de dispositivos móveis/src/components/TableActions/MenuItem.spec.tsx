import { Menu } from '@chakra-ui/react';

import { render, screen, fireEvent } from '../../helper/test/test-utils';
import MenuItem from "./MenuItem";

const MenuMenuItem = (props) => (
  <Menu><MenuItem {...props} /></Menu>
)

describe('Component MenuItem from Table Action', () => {
  it('should match with snapshot', () => {
    const { container } = render(<MenuMenuItem text="text" onClick={() => false} />);
    expect(container).toMatchSnapshot();
  });

  it('should render text prop', () => {
    render(<MenuMenuItem text="test" onClick={() => false} />);

    expect(screen.getByText('test')).toBeInTheDocument();
  });

  it('should run onClick function when click on component', () => {
    const onClick = jest.fn();
    render(<MenuMenuItem text="test" onClick={onClick} />);

    fireEvent.click(screen.getByText('test'));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
