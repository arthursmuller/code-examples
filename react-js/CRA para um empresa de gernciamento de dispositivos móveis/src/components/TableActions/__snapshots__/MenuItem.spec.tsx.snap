// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MenuItem from Table Action should match with snapshot 1`] = `
<div>
  <button
    class="chakra-menu__menuitem css-1jti93q"
    data-index="0"
    id="menu-list-2-menuitem-3"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    text
  </button>
</div>
`;
